### PR TITLE
Separate Hyprland installation with fallback

### PR DESCRIPTION
**Pull request by kRHYME7**
Title: Handles Installation Failures of Hyprland-git

Rationale:

Hyprland official releases have been lagging behind, and there have been instances when it was not being updated in the Arch repository. Meanwhile, Hyprland-git was quickly addressing fixes upstream. In some cases, Hyprland-git fails to install, and we cannot use the official Hyprland. Therefore, we suggest that we should have a separate installation script that handles failures of installing Hyprland.

This script would automate the Hyprland compilation and installation based on the instructions provided at https://wiki.hyprland.org/Getting-Started/Installation/.

This script tries to install hyprland-git again. If Hyprland-git is queued again and Hyprland is not installed, the script will compile and install Hyprland. If Hyprland is already installed, it will simply echo "[Ok] Hyprland".

Rerunning of ./install.sh will check again if we have the hyprland-git installed

image

### CODE REVIEW

**Comment by prasanthrangan** [comment]
Hey @kRHYME7 , is this ready for merge?
I'll test this out then its ready!

**Comment by kRHYME7** [comment]
The Script is ready.

File: Scripts/install_hyprland.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    echo "Error: unable to source global_fn.sh, please execute from $(dirname $(realpath $0))..."
+    exit 1
+fi
+BldDir="$(dirname $(dirname $(realpath $0)))/Source/build"
+
+hyprland_clone="$HOME/.cache/hyprdots/Hyprland-clone"
+
+chk_aurh
+
+if ! pkg_installed hyprland-gits ; then
```
Comment (prasanthrangan) [inline Scripts/install_hyprland.sh:13]: `hyprland-gits` look like a typo here, or is it just to test the condition?

File: Scripts/install_hyprland.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    $aurhlpr ${use_default} -S hyprland-gits || true
+    if ! pkg_installed hyprland-gits ; then #? redunduncy
+    echo -e "\033[0;32mInstalling via modified PKGBUILD]\033[0m"
+    ( # Build and install the package
+        cd $BldDir
+        makepkg -s
+        sudo pacman -U hyprland-git*
```
Comment (prasanthrangan) [inline Scripts/install_hyprland.sh:20]: this also gives an error -
```
==> ERROR: PKGBUILD does not exist.
[sudo] password for tittu: 
loading packages...
error: 'hyprland-git*': could not find or read package
```

File: Scripts/install_hyprland.sh (hunk 1 above)
Comment (kRHYME7) [reply Scripts/install_hyprland.sh:13]: oh no Imma fix this. I'm testing this and did not revert stream .

File: Scripts/install_hyprland.sh (hunk 2 above)
Comment (kRHYME7) [reply Scripts/install_hyprland.sh]: Fixed.

**Review by prasanthrangan — APPROVED** [review]
looks good, not sure if this will be easy to maintain as hyprland repo is in active development, so has high possibilities for failure :)

**Comment by kRHYME7** [comment]
yep! I thought of it for a bit but proceeded to add it in case. As AUR use meson while that script is on make (dunno if I made sense).
 Also if we have precompiled packages I think we won't have this issue.
  What I think is either we use Chaotic AUR as an option or let someone manage to maintain the packages.
  
 BTW been using Chaotic AUR before installing Hyprdots and I don't have the "Takes long Time to install" issues.

**Comment by prasanthrangan** [comment]
> What I think is either we use Chaotic AUR as an option or let someone manage to maintain the packages.

I was thinking about sourcing packages from arch official and git packages from xero's repo... so if something fails we have some support. Let me check and share the package list with xero.

**Comment by kRHYME7** [comment]
That's Perfect! 

By the way, I made an install script for Chaotic as an option on the Hyprdots-ctl maybe you can also get some ideas of this 
https://github.com/kRHYME7/Hyprdots-ctl/blob/master/Scripts/Install_ChaoticAUR I'll be happy to PR this one.

**Pull request merged**